controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java

```diff
@@ -71,23 +74,25 @@ public String getHandlerName() {
     }
 
     @Override
-    public boolean authenticate(String token) {
+    public Principal authenticate(String token) throws AuthException {
         String[] parts = parseToken(token);
         String userName = parts[0];
         String password = parts[1];
 
         try {
-            return userMap.containsKey(userName) && encryptor.checkPassword(password, userMap.get(userName).encryptedPassword);
+            if (userMap.containsKey(userName) && encryptor.checkPassword(password, userMap.get(userName).encryptedPassword)) {
+                return new UnixPrincipal(userName);
+            }
         } catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
             log.warn("Excpetion during password authentication", e);
-            return false;
+            throw new AuthenticationException(e);
         }
+        return null;
```

Comment:
What are the semantics of `null` here?   A good practice would be to avoid unnecessary information disclosure, e.g. whether a given user exists.

Reply:
Fixed. Throwing an exception when the user does not exist.